osl/preprocessing/batch.py

```diff
@@ -748,9 +762,15 @@ def run_proc_batch(
     if outdir is None:
         # Use the current working directory
         outdir = os.getcwd()
-    outdir = validate_outdir(outdir)
-    logsdir = validate_outdir(logsdir or outdir / "logs")
-    reportdir = validate_outdir(reportdir or outdir / "report")
+    if '{' in outdir and '}' in outdir:
+        # validate the parrent outdir - later do so for each subdirectory
+        tmpoutdir = validate_outdir(outdir.split('{')[0])
+        logsdir = validate_outdir(logsdir or tmpoutdir / "logs")
+        reportdir = validate_outdir(reportdir or tmpoutdir / "report")
```

Comment:
I think if we're writing subject data to separate directories this would create a separate report directory in each subject's directory: sub-{...}/report/sub-{...}/<figures>.png, we probably still want the report directory to be:
`reportdir = validate_outdir(reportdir or outdir / "report")`

Reply:
That is what is currently happening, because the report dir is written in `tmpoutdir`, which is the parent directory (without the {sub-} business)